From 47c0225fb0b46e6cecc38065c0491fe8c63d6a5f Mon Sep 17 00:00:00 2001
From: rnsbox <rnsbox@localhost>
Date: Wed, 14 Oct 2026 01:32:21 +0300
Subject: [PATCH 4/5] build: overridable -j and buildroot toolchain path

NPROC honors RNSBOX_NPROC and br-rootfs-pack honors
RNSBOX_BR_TOOLCHAIN_PATH, so board flows can cap parallelism and point
the buildroot userspace at the newer Arm GNU 11.3 aarch64 glibc
toolchain from host-tools while the kernel/u-boot keep building with the
vendor-pinned linaro 6.3.1.
---
 build/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/Makefile b/build/Makefile
index 190880827..60843b374 100644
--- a/build/Makefile
+++ b/build/Makefile
@@ -30,7 +30,7 @@ qstrip = $(strip $(subst ",,$(1)))
 ################################################################################
 # Default actions
 ################################################################################
-NPROC := $(shell nproc)
+NPROC := $(if $(RNSBOX_NPROC),$(RNSBOX_NPROC),$(shell nproc))
 
 export CHIP_ARCH_L := $(shell echo $(CHIP_ARCH) | tr A-Z a-z)
 export BORAD_FOLDER_PATH := ${BUILD_PATH}/boards/${CHIP_ARCH_L}/${PROJECT_FULLNAME}
@@ -560,7 +560,7 @@ br-rootfs-prepare:
 br-rootfs-pack:export TARGET_OUTPUT_DIR=$(BR_DIR)/output/$(BR_BOARD)
 br-rootfs-pack:
 	$(call print_target)
-	${Q}$(MAKE) -C $(BR_DIR) $(BR_DEFCONFIG) BR2_TOOLCHAIN_EXTERNAL_PATH=$(CROSS_COMPILE_PATH)
+	${Q}$(MAKE) -C $(BR_DIR) $(BR_DEFCONFIG) BR2_TOOLCHAIN_EXTERNAL_PATH=$(if $(RNSBOX_BR_TOOLCHAIN_PATH),$(RNSBOX_BR_TOOLCHAIN_PATH),$(CROSS_COMPILE_PATH))
 	${Q}$(MAKE) -j${NPROC} -C $(BR_DIR) source
 	${Q}$(MAKE) -j${NPROC} -C $(BR_DIR)
 	# ${Q}rm -rf $(BR_ROOTFS_DIR)/*
-- 
2.43.0

